src/api/PDFDocument.ts

```diff
@@ -701,27 +708,94 @@ export default class PDFDocument {
     return pdfImage;
   }
 
+  /**
+   * Embed a PDF page into this document.
+   *
+   * For example:
+   * ```js
+   * const pdfDoc = await PDFDocument.create();
+   *
+   * const sourcePdfUrl = 'https://pdf-lib.js.org/assets/with_large_page_count.pdf';
+   * const sourceBuffer = await fetch(sourcePdfUrl).then((res) => res.arrayBuffer());
+   * const sourcePdfDoc = await PDFDocument.load(sourceBuffer);
+   *
+   * // embed page 74 into pdfDoc
+   * const embeddedPage = await pdfDoc.embedPdfDocument(sourcePdfDoc,73);
+   * ```
+   *
+   * @param document The source PDF document that contains the page to be embedded
+   * @param pageIndex Optionally, the page index (starting at 0 for the first page) of the page to embed. Defaults to 0 (the first page).
+   * @returns Resolves with the embedded pdf page.
+   */
+  async embedPdfDocument(
+    document: PDFDocument,
+    pageIndex?: number,
+  ): Promise<EmbeddedPDFPage> {
```

Comment:
I'd like to rename this method to `embedPdf`, just for the sake of brevity. I also think it would be best if this method accepted a raw PDF (similar to [`PDFDocument.load`](https://pdf-lib.js.org/docs/api/classes/pdfdocument#static-load)). These changes would keep the API more aligned with the existing `embed*` methods. And if the user wants to embed pages from an actual `PDFDocument` object, they can simply use the `PDFDocument.embedPdfPage` method.

Also, this API doesn't handle embedding multiple pages from a document very efficiently. Doing so would likely result in many duplicated objects, due to a new `PDFObjectCopier` being created and used for each page the user wants to copy. This inefficiency can be avoided by creating one `PDFObjectCopier` and using it to copy all of the desired pages for a given document. Of course, to achieve this, the APIs will need to be converted to accept and return arrays. For example, this method's signature would become:
```js
async embedPdf(
  pdf: string | Uint8Array | ArrayBuffer, 
  indices: number[],
): Promise<EmbeddedPDFPage[]>
```

I should note that this is not just a theoretical concern. The initial API I created for copying pages in version v0.x.x of pdf-lib did not handle copying multiple pages either. Users ran into the duplicate objects issue I explained above. So I changed the [page copying API](https://pdf-lib.js.org/docs/api/classes/pdfdocument#copypages) to accept an array of indices in v1.0.0 of pdf-lib.

Reply:
I agrees, done! :)